hw8/task_1.py

```diff
@@ -63,3 +51,19 @@ def __setitem__(self, key: str, value: str):
 
     def __getitem__(self, key: str):
         return self.storage[key]
+
+    def key_name_checker(self, key: str) -> None:
+        if bool(
+            [
+                letter
+                for letter in key
+                if letter not in (*string.ascii_letters, "_", *string.digits)
+            ]
+        ):
```

Comment:
You don't need bool here.
You can use `all`. 
Use `set` instead of `tuple`.  (O(n) vs O(1) search)
It is good practice to extract set to a variable to avoid computing it on each key. 
```suggestion
if all(char in {*string.ascii_letters, "_", *string.digits} for char in key):
```